src/Client/Core/OrchestrationMetadata.cs

```diff
@@ -48,41 +36,43 @@ public sealed class OrchestrationMetadata
     /// <value>The unique ID of the orchestration instance.</value>
     public string InstanceId { get; }
 
+    public DataConverter? DataConverter { get; init; }
```

Comment:
We'll need XML doc comments on this new public property.

Reply:
But curious why you decided to make this public?

Reply:
I had to do some refactoring of this type to remove gRPC dependency and allow for public construction as it is used in the client project. So the client needs to provide the data converter here.